Snowflake.Data.Tests/UnitTests/ArrowResultSetTest.cs

```diff
@@ -92,25 +92,34 @@ public void TestHasRowsReturnsFalseIfNoRows()
             Assert.IsFalse(_arrowResultSet.HasRows());
         }
 
+        [Test]
+        public void TestRewindReturnsFalseBeforeFirstRow()
+        {
+            Assert.IsFalse(_arrowResultSet.Rewind());
+        }
+
         [Test]
         public void TestRewindReturnsFalseForFirstRow()
         {
+            _arrowResultSet.Next(); // move to first row
             Assert.IsFalse(_arrowResultSet.Rewind());
         }
 
         [Test]
         public void TestRewindReturnsTrueForSecondRowAndMovesToFirstRow()
         {
-            _arrowResultSet.Next();
+            _arrowResultSet.Next(); // move to first row
+            _arrowResultSet.Next(); // move to second row
             Assert.IsTrue(_arrowResultSet.Rewind());
             Assert.IsFalse(_arrowResultSet.Rewind());
         }
 
         [Test]
         public void TestRewindReturnsTrueForThirdRowAndMovesToFirstRow()
         {
-            _arrowResultSet.Next();
-            _arrowResultSet.Next();
+            _arrowResultSet.Next(); // move to first row
+            _arrowResultSet.Next(); // move to second row
+            _arrowResultSet.Next(); // move to third row
             Assert.IsTrue(_arrowResultSet.Rewind());
             Assert.IsTrue(_arrowResultSet.Rewind());
             Assert.IsFalse(_arrowResultSet.Rewind());
```

Comment:
cannot place comment below - should we do sth about ln 152 or can we get rid of the comment?